src/Compilers/Core/Portable/Collections/ImmutableArrayDictionary`2+Enumerator.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System.Collections;
+using System.Collections.Generic;
+
+namespace Microsoft.CodeAnalysis.Collections
+{
+    partial struct ImmutableArrayDictionary<TKey, TValue>
+    {
+        public struct Enumerator : IEnumerator<KeyValuePair<TKey, TValue>>, IDictionaryEnumerator
```

Comment:
>Enumerator [](start = 22, length = 10)

I didn't understand why provide our own `Enumerator` type (and `KeyCollection.Enumerator` and `ValueCollection.Enumerator`). Is there a problem with forwarding `GetEnumerator()` on the immutable wrapper types to the wrapped `GetEnumerator()` method?

Reply:
I'll investigate changing the enumerator(s) to use the ones provided by the wrapped dictionary.

Reply:
➡️ We can't change `KeyCollection.Enumerator` or `ValueCollection.Enumerator` without changing the performance characteristics of these collections (forced allocations that are otherwise avoidable). Removing only `Enumerator` didn't seem to offer substantial clarity for this implementation.